File: javascriptnotesfroprofessionals/java-backend-for-ajax/src/main/java/org/kd/server/config/RestTemplateConfig.java
package org.kd.server.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.web.client.RestTemplateBuilder;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.client.RestTemplate;

@Configuration
public class RestTemplateConfig {

    @Bean
    public RestTemplate countryRestTemplate(RestTemplateBuilder restTemplateBuilder) {
        return restTemplateBuilder.rootUri("http://localhost:8080").build();
    }

    @Bean
    public CommonUtility commonUtility(){
        return new CommonUtility();
    }

}

File: javascriptnotesfroprofessionals/java-backend-for-ajax/src/main/java/org/kd/server/config/RestTemplateConfig.kt
package org.kd.server.config

import org.kd.server.CommonUtility
import org.springframework.boot.web.client.RestTemplateBuilder
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.client.RestTemplate

@Configuration
class RestTemplateConfig {

    @Bean
    fun countryRestTemplate(restTemplateBuilder: RestTemplateBuilder): RestTemplate {
        return restTemplateBuilder.rootUri("http://localhost:8080").build()
    }

    @Bean
    fun commonUtility(): CommonUtility {
        return CommonUtility()
    }

}
